### ISSUE TITLE
Batch normalization with sub-batches

### ISSUE THREAD

**Issue by jingyan1216** [vlfeat/matconvnet#405]
Hi,

I was hoping to have a little clarification on how batch normalization operates with sub-batches during training time. Based on my understanding, the sub-batches are evaluated independently, and the gradients are accumulated over the sub-batches. Is it correct to say that batch normalization computes the mean and variance per sub-batch, instead of per batch?

Thanks!

**Comment by vedaldi** [comment]
Hi, yes, that is unavoidable unfortunately.
